build.assets/build-webassets-if-changed.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+#
+# Rebuilds a UI make target based on whether the given directories contents have changed
+# minus any node-modules.
+#
+
+set -eo pipefail
+
+ROOT_PATH="$(realpath "$(dirname "$0")/..")"
+MAKE="${MAKE:-make}"
+
+if ! command -v "$MAKE" >/dev/null; then
+  echo "Unable to find \"$MAKE\" on path."
+  exit 1
+fi
+
+if [ "$#" -lt 4 ]; then
+  echo "Usage: $0 <type> <last-sha-file> <build-target> <directories...>"
+  exit 1
+fi
+
+TYPE="$1"
+LAST_SHA_FILE="$ROOT_PATH/$2"
+BUILD_TARGET="$3"
+shift 3
+SRC_DIRECTORIES=("$@")
+
+# Calculate the current hash-of-hashes of the given source directories. Adds in yarn.lock and package.json as well.
+# This excludes node_modules, as the yarn.lock/package.json differences should handle this.
+CURRENT_SHA="$( (for dir in "${SRC_DIRECTORIES[@]}"; do
+  find "$ROOT_PATH/$dir" -type f
+done && echo "$ROOT_PATH/yarn.lock" && echo "$ROOT_PATH/package.json") | 
+  grep -v "node_modules" | xargs -I{} -n1 -P8 sha256sum {} | sort -k 2 | sha256sum |
```

Comment:
it looks like by default the `sha256sum` binary is not present on OSX 
```
 make ensure-webassets
/Users/marek/go/src/github.com/gravitational/teleport//build.assets/build-webassets-if-changed.sh: line 37: sha256sum: command not found
xargs: sha256sum: No such file or directory
make: *** [ensure-webassets] Error 127
```

Reply:
Great catch, looks like `shasum` is.

Reply:
Just made a change that selects from a variety of sha sum executables, let me know if this works.